mne/io/eeglab/eeglab.py

```diff
@@ -215,16 +235,30 @@ class RawEEGLAB(_BaseRaw):
         Path or instance of montage containing electrode positions.
         If None, sensor locations are (0,0,0). See the documentation of
         :func:`mne.channels.read_montage` for more information.
+    eog : list | tuple | 'auto'
+        Names or indices of channels that should be designated
+        EOG channels. If 'auto', the channel names containing
+        ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
+    event_id : dict | None
+        The ids of the events to consider. If None (default),
+        an empty dict is used and `event_id_func` (see below) is called on
+        every event value. If dict, the keys will be mapped to trigger values
+        on the stimulus channel and only keys not in `event_id` will be handled
+        by `event_id_func`.
+        Keys are case-sensitive.
+        Example: {'SyncStatus': 1; 'Pulse Artifact': 3}.
+    event_id_func : None | callable
+        What to do for events not found in `event_id`. Must
+        take one `str` argument and return an `int`. If `None`, defaults to
+        `strip_non_integer`, which strips event codes such as "D128" or "S  1"
+        of their non-integer parts and returns the integer. Any event that is
+        not in `event_id` and cannot be parsed with this function is dropped.
```

Comment:
Well eeglab triggers can be, and often are, arbitrary strings, but they also often are mappable to ints in predictable ways. 






On Wed, Mar 2, 2016 at 9:49 AM -0800, "Teon L Brooks" <notifications@github.com> wrote:












In mne/io/eeglab/eeglab.py:
> +        EOG channels. If 'auto', the channel names containing
> +        ``EOG`` or ``EYE`` are used. Defaults to empty tuple.
> +    event_id : dict | None
> +        The ids of the events to consider. If None (default),
> +        an empty dict is used and `event_id_func` (see below) is called on
> +        every event value. If dict, the keys will be mapped to trigger values
> +        on the stimulus channel and only keys not in `event_id` will be handled
> +        by `event_id_func`.
> +        Keys are case-sensitive.
> +        Example: {'SyncStatus': 1; 'Pulse Artifact': 3}.
> +    event_id_func : None | callable
> +        What to do for events not found in `event_id`. Must
> +        take one `str` argument and return an `int`. If `None`, defaults to
> +        `strip_non_integer`, which strips event codes such as "D128" or "S  1"
> +        of their non-integer parts and returns the integer. Any event that is
> +        not in `event_id` and cannot be parsed with this function is dropped.



I guess I don't follow why we don't just internally use the strip_non_integer. are there that many varieties of event representation? do they all have a number embedded?



—
Reply to this email directly or view it on GitHub.